src/components/Article.jsx

```diff
@@ -28,15 +28,38 @@ export default function Article() {
             setComments(data.comments)
         })
         .catch(err => console.log("error", err))
-    }, [article_id]) 
+    }, [article_id, comments]) 
+
+
+    let [commentInput, setCommentInput] = useState("")
+    let [commentError, setCommentError] = useState(false)
 
     if (loadingArticle) return <p>loading</p>
 
+    function submitComment(event) {
+        event.preventDefault()
+        axios.post(`https://nc-news-backend-shahbazrafi.herokuapp.com/api/articles/${article_id}/comments`, {username: username, body: commentInput})
+        .then(() => {setCommentError(false)})
+        .catch(() => {setCommentError(true)})
+    }
+
+
+
     return <>
         <p>Article {article_id}</p>
     <div className="flex">
         <ArticleInfo key={article.article_id} article={article}/>
         <div className="ArticleBody">{article.body}</div>
+    </div>
+        <form onSubmit={submitComment}>
+            <label>New Comment: </label>
+            <input value={commentInput} onChange={(event) => {setCommentInput(event.target.value)}}></input>
+            <button type="submit">Submit</button>
+        </form>
+        <p>{commentError === true? "There was an error" : null}</p>
+        <p>Comments: {comments ? comments.length : "loading"}</p>
+    <div className="flex">
```

Comment:
I'd really like you to separate this out into a comments component, please. This will also mean you can use just one useEffect in each component (move the comments request down to the comments component), and make your code so much more readable. It also means that we are staying true to separating out our concerns.